packages/@lwc/template-compiler/src/shared/ir.ts

```diff
@@ -5,24 +5,123 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
```

Comment:
I would recommend this file to be renamed to `ast.ts` as this change is sunsetting the concept of `ir` (Intermediary Representation) is favor of a public and stable AST.